### ISSUE TITLE
Exclude dev dependencies by default

### ISSUE THREAD

**Issue by ppumkin** [dotnet/sourcelink#167]
After including these dependencies in my package project

- SourceLink.Create.CommandLine 2.8.3
- Microsoft.SourceLink.Vsts.Git beta-63127-02
 
Now my consuming project i get the error

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		Unable to resolve dependency 'Microsoft.SourceLink.Vsts.Git'. Source(s) used: 'nuget.org', 'Kinder', 'Microsoft Visual Studio Offline Packages'.			0	
```

It shows under dependency

![image](https://user-images.githubusercontent.com/1249683/45549394-8fbfaa80-b81f-11e8-89bb-ca13482dbf77.png)

**This error originates on a Test project I have that is NET451** 

Does not affect CORE 2.1 / STANDARD 2.0

**Comment by ppumkin** [comment]
I added "all" to PrivateAssets to those packages - And my consuming projects no longer try and download these dependencies. Worth documenting somewhere?

**Comment by ctaggart** [comment]
Using `PrivateAssets="All"` is documented on the README.md is each example. Can you open a pull request for how you would make this clearer?

The other day, I noticed that when adding `xunit.runner.visualstudio` from Visual Studio, it was automatically set up like this:
```
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
    </PackageReference>
```
I'm curious if there is a way to do something similar.

**Comment by bording** [comment]
@ctaggart Those changes are from the new development dependency support added in NuGet 4.8, so if the package has that flag, it will be added like that now.

**Comment by ctaggart** [comment]
I see, C:\Users\taggac\.nuget\packages\xunit.runner.visualstudio\2.4.0\xunit.runner.visualstudio.nuspec has:
``` xml
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <developmentDependency>true</developmentDependency>
```

That is already in C:\Users\taggac\.nuget\packages\microsoft.sourcelink.github\1.0.0-beta-63127-02\microsoft.sourcelink.github.nuspec

**Comment by ppumkin** [comment]
> Using PrivateAssets="All" is documented on the README.md is each example.

Ohhhhhh..... I saw that but I had never used that before. So at the time of reading the documentation the first time I had no idea what it was for.

I also added the packages using NuGet and not editing the packages file. After not realising what that meant and searching around I found it on the Microsoft Documentation and added it via the properties pane on the package.

Here is a PR https://github.com/dotnet/sourcelink/pull/171
